src/state/megaStore.tsx

```diff
@@ -135,14 +147,23 @@ export const Provider: ParentComponent = (props) => {
                 if (state.setup_error) {
                     throw state.setup_error;
                 }
-                setState({ wallet_loading: true });
+                setState({
+                    wallet_loading: true,
+                    load_stage: "checking_double_init"
+                });
+                await doubleInitDefense();
+                setState({ load_stage: "downloading" });
+                await initializeWasm();
+                setState({ load_stage: "setup" });
                 const mutinyWallet = await setupMutinyWallet(settings);
                 // Get balance optimistically
-                const balance = await mutinyWallet.get_balance();
```

Comment:
is removing this get balance intended?

Reply:
oh I meant to ask about this... is this helping anything? i remove it and it doesn't seem to change much?

Reply:
fixed (I moved it down but yeah we do need it)